xray/core/dataset.py

```diff
@@ -612,11 +612,10 @@ def __getitem__(self, key):
         if utils.is_dict_like(key):
             return self.isel(**key)
 
-        key = np.asarray(key)
-        if key.ndim == 0:
-            return DataArray._new_from_dataset(self, key.item())
+        if hashable(key):
```

Comment:
Add a regression test for this? I think you identified a failing case in your last PR.